source/backend/tests/unit/api/Services/FinancialCodeServiceTest.cs

```diff
@@ -0,0 +1,173 @@
+            repo4.Setup(x => x.GetAllCostTypeCodes()).Returns(new List<PimsCostTypeCode>());
+            repo5.Setup(x => x.GetAllFinancialActivityCodes()).Returns(new List<PimsFinancialActivityCode>());
+            repo6.Setup(x => x.GetAllWorkActivityCodes()).Returns(new List<PimsWorkActivityCode>());
+            repo7.Setup(x => x.GetAllResponsibilityCodes()).Returns(new List<PimsResponsibilityCode>());
+
+            // Act
+            var result = service.GetAllFinancialCodes();
+
+            // Assert
+            result.Should().BeAssignableTo<IList<FinancialCodeModel>>();
+            repo1.Verify(x => x.GetAllBusinessFunctionCodes(), Times.Once);
+            repo2.Verify(x => x.GetAllChartOfAccountCodes(), Times.Once);
+            repo3.Verify(x => x.GetAllYearlyFinancialCodes(), Times.Once);
+            repo4.Verify(x => x.GetAllCostTypeCodes(), Times.Once);
+            repo5.Verify(x => x.GetAllFinancialActivityCodes(), Times.Once);
+            repo6.Verify(x => x.GetAllWorkActivityCodes(), Times.Once);
+            repo7.Verify(x => x.GetAllResponsibilityCodes(), Times.Once);
+        }
+
+        [Fact]
+        public void GetAll_NoPermission()
+        {
+            // Arrange
+            var user = PrincipalHelper.CreateForPermission();
+            var service = helper.Create<FinancialCodeService>(user);
+            var repo1 = helper.GetService<Mock<IBusinessFunctionCodeRepository>>();
+            var repo2 = helper.GetService<Mock<IChartOfAccountsCodeRepository>>();
+            var repo3 = helper.GetService<Mock<IYearlyFinancialCodeRepository>>();
+            var repo4 = helper.GetService<Mock<ICostTypeCodeRepository>>();
+            var repo5 = helper.GetService<Mock<IFinancialActivityCodeRepository>>();
+            var repo6 = helper.GetService<Mock<IWorkActivityCodeRepository>>();
+            var repo7 = helper.GetService<Mock<IResponsibilityCodeRepository>>();
+
+            repo1.Setup(x => x.GetAllBusinessFunctionCodes()).Returns(new List<PimsBusinessFunctionCode>());
+            repo2.Setup(x => x.GetAllChartOfAccountCodes()).Returns(new List<PimsChartOfAccountsCode>());
+            repo3.Setup(x => x.GetAllYearlyFinancialCodes()).Returns(new List<PimsYearlyFinancialCode>());
+            repo4.Setup(x => x.GetAllCostTypeCodes()).Returns(new List<PimsCostTypeCode>());
+            repo5.Setup(x => x.GetAllFinancialActivityCodes()).Returns(new List<PimsFinancialActivityCode>());
+            repo6.Setup(x => x.GetAllWorkActivityCodes()).Returns(new List<PimsWorkActivityCode>());
+            repo7.Setup(x => x.GetAllResponsibilityCodes()).Returns(new List<PimsResponsibilityCode>());
+
+            // Act
+            Action act = () => service.GetAllFinancialCodes();
+
+            // Assert
+            act.Should().Throw<NotAuthorizedException>();
+            repo1.Verify(x => x.GetAllBusinessFunctionCodes(), Times.Never);
```

Comment:
with the new pattern we are moving away from mocking/verifying when we are throwing an exception that would cause a mock not to be used.

Reply:
Can you please point to specific example of new pattern? I remember we talked about this but haven't seen the new pattern in action. In my tests I have tried to trim down the mocking/checking but seemed useful in this case to check that no underlying repos where called.

Reply:
have a look at my PR, most of the usages of Times.Never have been removed.